src/v/storage/probe.cc

```diff
@@ -16,7 +16,46 @@
 
 #include <seastar/core/metrics.hh>
 
+#include <type_traits>
+
 namespace storage {
+
+void node_probe::set_disk_metrics(
+  uint64_t total_bytes, uint64_t free_bytes, disk_space_alert alert) {
+    _disk = {
+      .total_bytes = total_bytes,
+      .free_bytes = free_bytes,
+      .space_alert = alert};
+}
+
+void node_probe::setup_node_metrics() {
+    if (config::shard_local_cfg().disable_metrics()) {
+        return;
+    }
+
+    namespace sm = ss::metrics;
+    _metrics.add_group(
+      prometheus_sanitize::metrics_name("storage:disk"),
+      {
+        sm::make_total_bytes(
+          "total_bytes",
+          [this] { return _disk.total_bytes; },
+          sm::description("Total size of attached storage, in bytes.")),
+        sm::make_total_bytes(
+          "free_bytes",
+          [this] { return _disk.free_bytes; },
+          sm::description("Disk storage bytes free.")),
+        sm::make_gauge(
+          "free_space_alert",
+          [this] {
+              return static_cast<std::underlying_type_t<disk_space_alert>>(
+                _disk.space_alert);
+          },
```

Comment:
nit:

since the values 0,1,2 effectively become part of the metrics interface a useful thing to do is define the translation here so that the enum values are irrelevant:

```c++
enum class disk_space_alert { ok, not_ok, very_bad };
...
        sm::make_gauge(
           "free_space_alert",
           [this] {
switch (_disk.space_alert) {
   case ok: return 1;
   case not_ok: return 2;
...
};
           },
```

then the underlying value is irrelevant.